include/mbedtls/ssl.h

```diff
@@ -1639,6 +1639,26 @@ struct mbedtls_ssl_context {
      */
     mbedtls_ssl_protocol_version MBEDTLS_PRIVATE(tls_version);
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+    /**
```

Comment:
I'm not sure how I feel about this.

On one hand, I'm not a big fan of fields whose interpretation depend on other fields. And the fact that we're playing tricks later like giving two names to the same constant is not really making me happier on that front.

On the other hand, we want don't want to use two distinct fields for server and client because a given context is only ever going to use one of those fields, so having two would be a waste of RAM.

Now that I write it this way, this seems exactly a case for using a `union`, with two fields inside, one for server one for client.

Wdyt?